packages/teuchos/core/src/Teuchos_TestingHelpers.hpp

```diff
@@ -258,6 +258,28 @@ bool compareFloatingArrays(
   Teuchos::FancyOStream &out
   );
 
+/** \brief Compare if two array objects are the same up to an absolute
+ * difference: elements <tt>a1[i]</tt> and <tt>a2[i]</tt> are considered
+ * the same if <tt>|a1[i]-a2[i]| <= tol</tt>.
+ *
+ * This function works with any two array objects are the same size and have
+ * the same element value types.  The function is templated on the container
+ * types and therefore can compare any two objects that have size() and
+ * operator[](i) defined. <tt>Teuchos::ScalarTraits</tt> must have a specialization
+ * for the element type.
+ *
+ * \returns Returns <tt>true</tt> if the compare and <tt>false</tt> otherwise.
```

Comment:
“If the compare” seems a bit weird. Could it be “if the comparison holds”?

Reply:
I agree, I copy-pasted that but it doesn't make sense. For the new function I just spelled out the formula for exactly what's being compared, since I think that's readable enough.